master: build-environment on WCOSS and file-name change

chan-hoo opened the pull request:
## DESCRIPTION OF CHANGES: 
- Add the build-environment files for WCOSS Dell and Cray.
- The name and extension of the build-environment file ('README_xxx.txt') are changed to 'build_xxx.env' and the files are moved to a new directory 'ufs-srweather-app/env'.
- For users' convenience, the python environment for the workflow is put in a new file 'env/wflow_xxx.env'.

## TESTS CONDUCTED: 
- The SRW App ('master' branch) was built without any issues on WCOSS Dell/Cray as well as Hera.

## ISSUE: 
- Fixes the issue mentioned in #82 
- Related to https://github.com/NOAA-EMC/regional_workflow/issues/389

## CONTRIBUTORS: 
@BenjaminBlake-NOAA

JeffBeck-NOAA commented:
@mkavulich, can you take a look at this PR to confirm whether you're on board with the README file changes proposed here?    Thanks.

chan-hoo commented:
@JeffBeck-NOAA @mkavulich, if you agree with the README file changes, I'll merge the PR (https://github.com/NOAA-EMC/regional_workflow/pull/390), and update the 'Externals.cfg' file.

mkavulich commented on Externals.cfg:
```diff
@@ -1,9 +1,9 @@
 [regional_workflow]
 protocol = git
-repo_url = https://github.com/NOAA-EMC/regional_workflow
+repo_url = https://github.com/chan-hoo/regional_workflow
```
Sorry for the late review here...please remove the changes to Externals.cfg, these should not be committed to master

mkavulich commented on Externals.cfg:
```diff
@@ -1,9 +1,9 @@
 [regional_workflow]
```
Sorry for the late review here...please remove these Externals.cfg changes from your PR, these should not be committed to master

mkavulich commented on env/wflow_wcoss_dell_p3.env:
```diff
@@ -0,0 +1,5 @@
+# Python environemnt for workflow on WCOSS_dell_p3
```
Please correct the misspelling in the comments of these "wflow" files (environemnt -> environment)

christinaholtNOAA commented:
I was just dropping in to check out the renaming of some files, which piqued my attention due to other workflow activities.

I would suggest adding some documentation to the README file about how you would use the new .env files that you've added here for the configuration environments. I believe the existing documentation only points users to load the pre-existing build environment, but says nothing about loading a different environment before running the configuration layer.

chan-hoo commented:
@christinaholtNOAA @mkavulich, I've added a new readme file named 'RUNTIME' to the `ufs-srweather-app/docs' directory. I've also modified 'Externals.cfg' and corrected the typos. Please review them again.

christinaholtNOAA commented:
Thanks, @chan-hoo. Looks great. :)

mkavulich reviewed (approved):
Thank you for the quick turnaround @chan-hoo, the changes look good!

jwolff-ncar commented:
@chan-hoo Would you have time to cherry pick the file name changes from README to env into the release branch as well? This will make documentation significantly more straightforward between the two. If you don't have time, let me know so we can create separate documentation (quick start guide) for release vs develop.

chan-hoo commented:
@jwolff-ncar, no problem. I'll open a PR for this soon.
